src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BatchExecution.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.unit.statement;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import java.sql.BatchUpdateException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+import org.opentest4j.TestAbortedException;
+
+import com.microsoft.sqlserver.jdbc.SQLServerStatement;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.DBConnection;
+
+/**
+ * Tests batch execution with AE On connection
+ *
+ */
+@RunWith(JUnitPlatform.class)
+public class BatchExecution extends AbstractTest {
+
+    static Statement stmt = null;
+    static Connection connection = null;
+    static PreparedStatement pstmt = null;
+    static PreparedStatement pstmt1 = null;
+    static ResultSet rs = null;
+
+    @Test
+    public void testBatchExceptionAEOn() throws Exception {
+        testAddBatch1();
+        testExecuteBatch1();
+    }
+
+    public void testAddBatch1() {
```

Comment:
Suggestion: Adding javadocs will give hint to developers related to what we are going to test.